packages/server/src/index.ts

```diff
@@ -533,24 +533,26 @@ const routes = async (fastifyInstance: FastifyInstance) => {
   });
 
   interface JobExecuteParams {
-    deviceIdsOrOrigins: string;
+    deviceIdsOrOrigins?: string;
     jobId: string;
   }
 
-  fastifyInstance.post<{ Params: JobExecuteParams }>(
-    '/api/job/execute/:jobId/:deviceIdsOrOrigins',
+  fastifyInstance.post<{ Params: JobExecuteParams; Body: { deviceIdsOrOrigins: string[] | number[] } }>(
+    '/api/job/execute/:jobId/:deviceIdsOrOrigins?',
     async (req, reply) => {
-      const deviceIdsOrOrigins = req.params.deviceIdsOrOrigins.split(',');
-      const jobId = req.params.jobId;
+      const { deviceIdsOrOrigins, jobId } = req.params;
       const job = jobs.getById(jobId);
 
       if (!job) {
         reply.code(404).send({ status: 'error', error: 'Job not found' });
-
         return;
       }
 
-      const devices: DeviceControlConnection[] = deviceIdsOrOrigins
+      // If deviceIdsOrOrigins is not set in URL params, use the value from the request body
+      const deviceIdsOrOriginsList =
+        deviceIdsOrOrigins !== undefined ? deviceIdsOrOrigins.split(',') : req.body.deviceIdsOrOrigins;
```

Comment:
Are you supporting both the URL params and the body because you didn't want to make a breaking change?

Reply:
exactly.. I changed the API call within Rotom client.. But some tools rely on the API, so I thought it would be nice to have it backward compatible

Reply:
would you prefer a new endpoint instead? of good?